acme/acme/standalone.py

```diff
@@ -49,12 +56,94 @@ class ACMEServerMixin:  # pylint: disable=old-style-class
     allow_reuse_address = True
 
 
+class BaseDualNetworkedServers(object):
+    """Base class for a pair of IPv6 and IPv4 servers that tries to do everything
+       it's asked for both servers, but where failures in one server don't
+       affect the other.
+
+       If two servers are instantiated, they will serve on the same port.
+       """
+
+    def __init__(self, *args, **kwargs):
+        address = args[0]
+        port = address[1]
+        ServerClass = kwargs.pop("server_class", object)
+        self.threads = []
+        self.servers = []
+
+        # Must try True first.
+        # Ubuntu, for example, will fail to bind to IPv4 if we've already bound
+        # to IPv6. But that's ok, since it will accept IPv4 connections on the IPv6
+        # socket. On the other hand, FreeBSD will successfully bind to IPv4 on the
+        # same port, which means that server will accept the IPv4 connections.
+        # If Python is compiled without IPv6, we'll error out but (probably) successfully
+        # create the IPv4 server.
+        for ip_version in [True, False]:
+            try:
+                kwargs["ipv6"] = ip_version
+                new_address = (address[0],) + (port,) + address[2:]
+                new_args = (new_address,) + args[1:]
+                server = ServerClass(*new_args, **kwargs)
```

Comment:
ok but the linter thinks it's bad for readability but really we want it for functionality so I'm going to tell it to mind its own business.